src/main/java/org/betonquest/betonquest/variables/RandomNumberVariable.java

```diff
@@ -0,0 +1,123 @@
+package org.betonquest.betonquest.variables;
+
+import org.betonquest.betonquest.Instruction;
+import org.betonquest.betonquest.VariableNumber;
+import org.betonquest.betonquest.api.Variable;
+import org.betonquest.betonquest.api.config.quest.QuestPackage;
+import org.betonquest.betonquest.api.profiles.Profile;
+import org.betonquest.betonquest.exceptions.InstructionParseException;
+import org.betonquest.betonquest.exceptions.QuestRuntimeException;
+
+import java.text.DecimalFormat;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * This variable resolves into a random value (inclusive arguments)
+ * Note that it will return a different value for each call.
+ */
+public class RandomNumberVariable extends Variable {
+
+    /**
+     * Used for check if fractional uses limited decimal places
+     */
+    private static final int DOUBLE_LENGTH = "double".length();
+    /**
+     * The lower bar of the random amount
+     */
+    private final VariableNumber low;
+    /**
+     * The higher bar of the random amount
+     */
+    private final VariableNumber high;
+    /**
+     * If the value should be returned as {@code double}
+     */
+    private final boolean fractional;
+    /**
+     * The digit amount to round to in fractional mode stored in pattern
+     */
+    private final DecimalFormat format;
+
+    /**
+     * Creates a new {@link RandomNumberVariable}
+     *
+     * @param instruction the {@link Instruction} used to create the Variable
+     * @throws InstructionParseException When an error occurs
+     */
+    public RandomNumberVariable(final Instruction instruction) throws InstructionParseException {
+        super(instruction);
+        final String type = instruction.next();
+        if ("int".equalsIgnoreCase(type)) {
+            this.fractional = false;
+            this.format = null;
+        } else if (type.startsWith("double")) {
+            this.fractional = true;
+            this.format = getFormat(type);
+        } else {
+            throw new InstructionParseException(String.format("Unknown argument type: '%s'", type));
+        }
+
+        final QuestPackage pack = instruction.getPackage();
+
+        final String low = instruction.next();
+        this.low = new VariableNumber(pack, low.startsWith("{") ?
+                parseToVariableString(instruction) : low.replace(',', '.'));
+
+        final String high = instruction.next();
+        this.high = new VariableNumber(pack, high.startsWith("{") ?
+                parseToVariableString(instruction) : high.replace(',', '.'));
```

Comment:
More confusion about, if the variables should use a dot or a comma (Sure i know the answer, but thats exactly the issue).

Reply:
The logic for this is changed in the last Update.